src/Container.js

```diff
@@ -0,0 +1,334 @@
+   * @memberof Container
+   */
+  _internalOpen(userPath, { singlePlay = false, playOptions = null } = {}) {
+    const options = { singlePlay, playOptions };
+    this.reset();
+
+    const err = Features.basic(); // TODO Features.basic()
+    if (err) {
+      return this.client.trigger('unsupported');
+    }
+
+    this.loading = true;
+
+    this.initClient();
+
+    this.plugins.forEach(plugin => plugin.open(this));
+
+    let path = userPath;
+    if (null !== options.playOptions) {
+      const playOptionsQueryString =
+        'playOptions=' +
+        encodeURIComponent(JSON.stringify(options.playOptions));
+
+      path =
+        -1 === userPath.indexOf('?')
+          ? `${userPath}?${playOptionsQueryString}`
+          : `${userPath}&${playOptionsQueryString}`;
+    }
+
+    this.main.classList.add('loading');
+    this.main.setAttribute('src', path);
+
+    this.client.respond('singlePlay', options.singlePlay);
+    this.client.respond('playOptions', options.playOptions);
+    this.client.trigger('open');
+  }
+
+  /**
+   *
+   *
+   * @param {*} path
+   * @param {*} [options={}]
+   * @param {*} [playOptions={}]
+   * @memberof Container
+   */
+  openPath(path, options = {}, playOptions = {}) {
+    // This should be deprecated, support for old function signature
+    if ('boolean' === typeof options) {
+      options = {
+        singlePlay: false,
+        playOptions: playOptions
+      };
+    }
+    this._internalOpen(path, options);
+  }
+
+  /**
+   * Destroy and don't use after this
+   * @memberof Container
+   */
+  destroy() {
+    this.reset();
+    // Destroy in the reverse priority order
+    this.plugins
+      .slice()
+      .reverse()
+      .forEach(plugin => plugin.teardown(this));
+
+    this.main = null;
+    this.options = null;
+    this.dom = null;
+  }
+
+  /**
+   * Tell the application to start closing
+   * @memberof Container
+   */
+  close() {
+    if (this.loading || this.loaded) {
+      this.plugins.forEach(plugin => plugin.close(this));
+      this.client.trigger('close');
+      // Start the close
+      this.client.send('close');
+    } else {
+      this.reset();
+    }
+  }
+
+  /**
+   * The current version of SpringRollContainer
+   * @readonly
+   * @static
+   * @memberof Container
+   */
+  static get version() {
+    return 'CONTAINER_VERSION';
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {object|function} plugin your plugin. This will be merged with a base plugin to make sure your plugin has certain functions
+   * @memberof Container
+   */
+  static uses(plugin = BasePlugin) {
+    //Merging with the base plugin to guarantee we will have certain functions
+    const isConstructor = () =>
+      !!plugin.prototype &&
+      plugin.prototype.constructor &&
+      /^class/.test(plugin.prototype.constructor);
+
+    if ('function' !== typeof plugin && !isConstructor()) {
+      return;
+    }
+
+    PLUGINS.push(plugin);
+    PLUGINS.sort((a, b) => b.priority - a.priority);
```

Comment:
Also, I'm working on this dependency resolution mechanism (essentially topological sort), so I can forklift that over too once that's in a good place.